pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataCache.java

```diff
@@ -128,6 +130,24 @@ public interface MetadataCache<T> {
      */
     CompletableFuture<Void> create(String path, T value);
 
+    /**
+     * Create or update the value of the given path in the metadata store without version comparison.
+     * <p>
+     * This method is equivalent to
+     * {@link org.apache.pulsar.metadata.api.extended.MetadataStoreExtended#put(String, byte[], Optional, EnumSet)} or
+     * {@link MetadataStore#put(String, byte[], Optional)} if the metadata store does not support this extended API,
+     * with `Optional.empty()` as the 3rd argument. It means if the path does not exist, it will be created. If the path
+     * already exists, the new value will override the old value.
+     * </p>
+     * @param path the path of the object in the metadata store
+     * @param value the object to put in the metadata store
+     * @param options the create options if the path does not in the metadata store
+     * @return the future that indicates if this operation failed, it could fail with
+     *   {@link java.io.IOException} if the value failed to be serialized
+     *   {@link MetadataStoreException} if the metadata store operation failed
+     */
+    CompletableFuture<Void> put(String path, T value, EnumSet<CreateOption> options);
```

Comment:
```suggestion
    CompletableFuture<Void> put(String path, T value, Set<CreateOption> options);
```
It's better to depend on the `Set` interface instead of a particular implementation (`EnumSet`)

Reply:
Oh it looks like `EnumSet` is already in `MetadataStoreExtended`. Not really addressable.

Reply:
It makes sense to me since `MetadataCache` is implemented to hide some details to process when using `MetadataStore` directly.

Reply:
> It makes sense to me since `MetadataCache` is implemented to hide some details to process when using `MetadataStore` directly.

My comment was about using `EnumSet` in interface methods. It's not a good practice in interface design to require a specific implementation. Using `EnumSet` is similar to using `ArrayList` in interface methods. 
In this case it's not addressable since the `MetadataStoreExtended` already did that mistake and we don't want to start changing that as part of this PR.